secure/storage/src/namespaced_storage.rs

```diff
@@ -32,8 +32,8 @@ impl<T: KVStorage> KVStorage for NamespacedStorage<T> {
     }
 }
 
-impl<T> NamespacedStorage<T> {
-    pub fn new(storage: T, namespace: String) -> Self {
+impl NamespacedStorage {
+    pub fn new(storage: Box<dyn KVStorage>, namespace: String) -> Self {
```

Comment:
we could perhaps make new<T: KVStorage> and then box internally ... not sure one is better than the other. Thoughts?

Reply:
That's impossible: if `NamespacedStorage` takes a generic parameter (i.e. recursion), then so does `BoxedStorage`. Then the `impl From<SecureBackend> for BoxedStorage` becomes an `impl<T> From<SecureBackend> for BoxedStorage<T>`, but returning this `<T>` from an unknown runtime content is not possible / nonsensical.